pkg/volume/cinder/attacher.go

```diff
@@ -278,12 +279,22 @@ func (detacher *cinderDiskDetacher) Detach(deviceMountPath string, nodeName type
 		instanceid = instanceid[(ind + 1):]
 	}
 
+	pending, volumeStatus, err := detacher.cinderProvider.OperationPending(volumeID)
+	if err != nil {
+		return err
+	}
+	if pending {
+		glog.Infof("status of volume %q is %s, wait it to be finished", volumeID, volumeStatus)
+		return volume.ErrNotFinish
```

Comment:
Yeah, I will do it.
By the way, should we keep the old behavior or change it to return an error at Line 143 / 91 / 297 ?